jobs/multiqc.py

```diff
@@ -0,0 +1,197 @@
+
+
+def multiqc(
+    b: Batch,
+    dataset: Dataset,
+    tmp_prefix: Path,
+    paths: list[Path],
+    out_json_path: Path,
+    out_html_path: Path,
+    out_html_url: str | None = None,
+    out_checks_path: Path | None = None,
+    label: str | None = None,
+    ending_to_trim: set[str] | None = None,
+    modules_to_trim_endings: set[str] | None = None,
+    job_attrs: dict | None = None,
+    sample_id_map: dict[str, str] | None = None,
+    extra_config: dict | None = None,
+) -> list[Job]:
+    """
+    Run MultiQC for the files in `qc_paths`
+    @param b: batch object
+    @param tmp_prefix: bucket for tmp files
+    @param paths: file bucket paths to pass into MultiQC
+    @param dataset: Dataset object
+    @param out_json_path: where to write MultiQC-generated JSON file
+    @param out_html_path: where to write the HTML report
+    @param out_html_url: URL corresponding to the HTML report
+    @param out_checks_path: flag indicating that QC checks were done
+    @param label: To add to the report's, Batch job's, and Slack message's titles
+    @param ending_to_trim: trim these endings from input files to get sample names
+    @param modules_to_trim_endings: list of modules for which trim the endings
+    @param job_attrs: attributes to add to Hail Batch job
+    @param sample_id_map: sample ID map for bulk sample renaming:
+        (https://multiqc.info/docs/#bulk-sample-renaming-in-reports)
+    @param extra_config: extra config to pass to MultiQC
+    @return: job objects
+    """
+    title = 'MultiQC'
+    if label:
+        title += f' [{label}]'
+
+    mqc_j = b.new_job(title, (job_attrs or {}) | dict(tool='MultiQC'))
+    mqc_j.image(image_path('multiqc'))
+    STANDARD.set_resources(mqc_j, ncpu=16)
+
+    file_list_path = tmp_prefix / 'multiqc-file-list.txt'
+    with file_list_path.open('w') as f:
+        f.writelines([f'{p}\n' for p in paths])
+    file_list = b.read_input(str(file_list_path))
+
+    endings_conf = ', '.join(list(ending_to_trim)) if ending_to_trim else ''
+    modules_conf = (
+        ', '.join(list(modules_to_trim_endings)) if modules_to_trim_endings else ''
+    )
+
+    if sample_id_map:
+        sample_map_path = tmp_prefix / 'rename-sample-map.tsv'
+        _write_sample_id_map(sample_id_map, sample_map_path)
+        sample_map_file = b.read_input(str(sample_map_path))
+    else:
+        sample_map_file = None
+
+    if extra_config:
+        serialised = ', '.join(f'{k}: {v}' for k, v in extra_config.items())
+        extra_config_param = f'--cl-config "{serialised}"'
+    else:
+        extra_config_param = ''
+
+    report_filename = 'report'
+    cmd = f"""\
+    mkdir inputs
+    cat {file_list} | gsutil -m cp -I inputs/
+    
+    # Temporary fix for Somalier module before https://github.com/ewels/MultiQC/pull/1670
+    # is merged and released:
+    git clone --single-branch --branch fix-somalier https://github.com/vladsaveliev/MultiQC
+    pip3 install -e MultiQC
```

Comment:
Is this still necessary? I've noticed that this PR got merged, but not sure whether it also got released.

Reply:
It's not released yet